### ISSUE TITLE
If `diagnosticSeverityOverrides` is set, `typeCheckingMode` is ignored and set to "all".

### ISSUE THREAD

**Issue by kiyoon** [DetachHead/basedpyright#329]
I usually set `typeCheckingMode` to "standard" and customise some of the diagnostics using `diagnosticSeverityOverrides`. However, in recent version of basedpyright I see too many errors and I think something has changed compared to earlier version.

When I set like below, I think all type checking is turned on although I set it to "standard".

```lua
  settings = {
    basedpyright = {
      analysis = {
        typeCheckingMode = "standard",
        autoSearchPaths = true,
        useLibraryCodeForTypes = true,
        autoImportCompletions = true,
        diagnosticsMode = "openFilesOnly",
        diagnosticSeverityOverrides = {
          reportUnusedImports = false,
        },
      },
    },
  },
```

And if I remove `diagnosticSeverityOverrides`, then I see most errors being turned off.

**Comment by kiyoon** [comment]
Hi, I'm sorry for the wrong report. I think this is just an issue with my neovim settings. With adding some settings I noticed they just don't get passed to basedpyright. I will close the issue soon when I find the error for sure.

Sorry again!

**Comment by DetachHead** [comment]
no worries. you may have to restart the language server for changes to take effect, but i'm not sure though as i don't use neovim

**Comment by kiyoon** [comment]
I have to conclude that it's a bug in neovim, because it was returning a completely ridiculous option table randomly (like 30% of the time), thus making the settings default. I solved the problem anyway and it was definitely not due to basedpyright. Thank you!

**Comment by angbongon** [comment]
> I have to conclude that it's a bug in neovim, because it was returning a completely ridiculous option table randomly (like 30% of the time), thus making the settings default. I solved the problem anyway and it was definitely not due to basedpyright. Thank you!

@kiyoon how did you fix this?? I think I'm having the same issue right now

**Comment by kiyoon** [comment]
@angbongon My settings is a lua module that looks like this.

```lua
-- NOTE: in nvim 0.9.5, if you import this module at the bottom,
-- it will sometimes not work, returning `lang` instaed of `M`.
-- Super strange, likely a bug in nvim.
local lang = require("kiyoon.lang").lang

M = {
  settings = {
    -- python = {
    basedpyright = {
      analysis = {
        typeCheckingMode = "standard", -- off, basic, standard, strict, all
        autoSearchPaths = true,
        useLibraryCodeForTypes = true,
        autoImportCompletions = true,
        diagnosticsMode = "openFilesOnly", -- workspace, openFilesOnly
        diagnosticSeverityOverrides = {
          reportUnusedImports = false,
          reportUnusedVariable = false,
        },
      },
    },
  },
}

-- NOTE: pyright works with es-ES but not with es_ES. Weird.
if lang == "es" then
  M.cmd_env = { LC_ALL = "es-ES.UTF-8" }
elseif lang == "fr" then
  M.cmd_env = { LC_ALL = "fr-FR.UTF-8" }
elseif lang == "pt-br" then
  M.cmd_env = { LC_ALL = "pt-br.UTF-8" }
elseif lang == "pt-pt" then
  M.cmd_env = { LC_ALL = "pt-pt.UTF-8" }
end

return M
```

`kiyoon.lang` is also a simple lua module.

```lua
-- check environment variable

M = {}

local lang_env = os.getenv "LANG"

if lang_env == nil then
  M.lang = "en"
else
  if vim.startswith(lang_env, "es") then
    M.lang = "es"
  elseif vim.startswith(lang_env, "fr") then
    M.lang = "fr"
  elseif vim.startswith(lang_env, "pt_BR") then
    M.lang = "pt-br"
  else
    M.lang = "en"
  end
end

return M
```

If you move the `require("kiyoon.lang")` to the bottom of the `M = {}` statement, randomly you get lang instead of M when returning the module. So I think it's a bug in LuaJIT or something like that.
